lib/lumberjack/server.rb

```diff
@@ -42,31 +42,42 @@ def initialize(options={})
       @ssl.key = OpenSSL::PKey::RSA.new(File.read(@options[:ssl_key]),
                                         @options[:ssl_key_passphrase])
       @ssl_server = OpenSSL::SSL::SSLServer.new(@tcp_server, @ssl)
+      @stop = false
     end # def initialize
 
     def run(&block)
-      while true
-        connection = accept
-        Thread.new(connection) do |connection|
+      while !@stop
+        sock = @tcp_server.accept
+        Thread.new(sock) do |sock|
+          connection = accept_ssl(sock)
           connection.run(&block)
         end
       end
     end # def run
 
-    def accept(&block)
+    def accept_ssl(socket)
+      tries = 0
       begin
-        fd = @ssl_server.accept
+        ssl = OpenSSL::SSL::SSLSocket.new(socket, @ssl)
```

Comment:
`ssl` vs `@ssl`. Can you rename `@ssl` to `@ssl_context` to avoid naming confusion?